anchore_engine/utils.py

```diff
@@ -218,6 +218,39 @@ def shellcheck(x):
     return [x for x in cmd_list if shellcheck(x)]
 
 
+def run_piped_command_list(
+    cmd_lists, stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs
+):
+    """
+    Executes a series of subcommands from cmd_lists, the stdout result of each being piped stdin for the next.
+    For example, the command: 'echo {}' | grype
+    cmd_lists should be passed as: [['echo', '{}'], ['grype']]
+    Do not include the actual pipe symbol, it is inferred from the data structure.
+    """
+    if cmd_lists:
```

Comment:
in what situation the `cmd_lists` wouldn't be passed in? why is it useful to allow this?

Reply:
As you said below, raising an exception is more apropriate. The only scenario is user error, the whole goal of protecting this block was to prevent an NPE on output.communicate()

Reply:
So I will make that change as well